File: src/algorithms/min-operations.js
/**
 * Given a binary string boxes of length n, where boxes[i] is '0' if the ith 
 * box is empty, and '1' if it contains one ball, returns an array answer of 
 * size n, where answer[i] is the minimum number of operations needed to move 
 * all the balls to the ith box.
 * 
 * In one operation, one ball can be moved from a box to an adjacent box. Each 
 * answer[i] is calculated considering the initial state of the boxes.
 * 
 * @param {string} boxes - A string representing the number of balls in boxes
 * @returns {number[]} - Returns the number of operations to move all balls to
 *     each box from their initial starting places
 */
function minOperations (boxes) {
  let answer = [];
  let i = 0;
  let j = 1;
  let operations = 0;
  while (i < boxes.length) {
    if (boxes[j] === '1' && j !== i) {
      operations += Math.abs(i - j);
    }
    j++;
    if (j >= boxes.length) {
      answer.push(operations);
      i++;
      j = 0;
      operations = 0;
    }
  }
  return answer;
}

File: src/algorithms/min-operations.py
def min_operations(nums):
    """Returns the minimum number of operations to make nums strictly increasing.

    You are given an integer array nums (0-indexed). In one operation, you can choose an element of the array and increment it by 1. For example, if nums = [1,2,3], you can choose to increment nums[1] to make nums = [1,3,3].Return the minimum number of operations needed to make nums strictly increasing. An array nums is strictly increasing if nums[i] < nums[i+1] for all 0 <= i < nums.length - 1. An array of length 1 is trivially strictly increasing.

    Args:
        nums (List[int]): A list of positive integers.

    Returns:
        Integer: Returns a count of operations to make nums strictly increasing.
    """
    count = 0
    for i in range(1, len(nums)):
        if nums[i] <= nums[i - 1]:
            increment = nums[i - 1] - nums[i] + 1
            nums[i] += increment
            count += increment
    return count
